lutris/util/graphics/drivers.py

```diff
@@ -2,83 +2,139 @@
 
 Everything in this module should rely on /proc or /sys only, no executable calls
 """
+import contextlib
 import os
 import re
+from typing import Dict, Iterable, List
 
+from lutris.util.graphics.glxinfo import GlxInfo
 from lutris.util.log import logger
+from lutris.util.system import read_process_output
 
 MIN_RECOMMENDED_NVIDIA_DRIVER = 415
 
 
-def get_nvidia_driver_info():
+def get_nvidia_driver_info() -> Dict[str, Dict[str, str]]:
     """Return information about NVidia drivers"""
     version_file_path = "/proc/driver/nvidia/version"
     if not os.path.exists(version_file_path):
-        return
-    with open(version_file_path, encoding='utf-8') as version_file:
-        content = version_file.readlines()
-        nvrm_version = content[0].split(': ')[1].strip().split()
-        if 'Open' in nvrm_version:
+        return {}
+    with contextlib.suppress(OSError):
+        with open(version_file_path, encoding="utf-8") as version_file:
+            content = version_file.readlines()
+        nvrm_version = content[0].split(": ")[1].strip().split()
+        if "Open" in nvrm_version:
             return {
-                'nvrm': {
-                    'vendor': nvrm_version[0],
-                    'platform': nvrm_version[1],
-                    'arch': nvrm_version[6],
-                    'version': nvrm_version[7]
+                "nvrm": {
+                    "vendor": nvrm_version[0],
+                    "platform": nvrm_version[1],
+                    "arch": nvrm_version[6],
+                    "version": nvrm_version[7],
                 }
             }
         return {
-            'nvrm': {
-                'vendor': nvrm_version[0],
-                'platform': nvrm_version[1],
-                'arch': nvrm_version[2],
-                'version': nvrm_version[5],
-                'date': ' '.join(nvrm_version[6:])
+            "nvrm": {
+                "vendor": nvrm_version[0],
+                "platform": nvrm_version[1],
+                "arch": nvrm_version[2],
+                "version": nvrm_version[5],
+                "date": " ".join(nvrm_version[6:]),
             }
         }
+    # If the /proc file failed, look it up with glxinfo.
+    glx_info = GlxInfo()
+    platform = read_process_output(["uname", "-s"])
+    arch = read_process_output(["uname", "-m"])
+    return {
+        "nvrm": {
+            "vendor": glx_info.opengl_vendor,  # type: ignore[attr-defined]
+            "platform": platform,
+            "arch": arch,
+            "version": glx_info.opengl_version.rsplit(maxsplit=1)[-1],  # type: ignore[attr-defined]
+        }
+    }
 
 
-def get_nvidia_gpu_ids():
+def get_nvidia_gpu_ids() -> List[str]:
     """Return the list of Nvidia GPUs"""
-    return os.listdir("/proc/driver/nvidia/gpus")
+    with contextlib.suppress(OSError):
+        return os.listdir("/proc/driver/nvidia/gpus")
+    values = read_process_output(
```

Comment:
**Gah!** again.

This one is very clear; you are writing ``try: except OSError:`` in an obscure way. Just use ``try: except OSError:`` here.

I'm not sure we need a log message at every place we catch this- as long as the first one is logged. Which of these runs first, I wonder?